src/Features/Core/Portable/QuickInfo/CommonSemanticQuickInfoProvider.cs

```diff
@@ -172,12 +172,37 @@ bool TryGetGroupText(SymbolDescriptionGroups group, out ImmutableArray<TaggedTex
             void AddSection(string kind, ImmutableArray<TaggedText> taggedParts)
                 => sections.Add(QuickInfoSection.Create(kind, taggedParts));
 
+            // We show a special message if the Task being awaited has no return
+            if (tokenInformation.ShowAwaitReturn && (tokenInformation.Symbols.First() as INamedTypeSymbol)?.SpecialType == SpecialType.System_Void)
+            {
+                var builder = ImmutableArray.CreateBuilder<TaggedText>();
+                builder.AddText(FeaturesResources.Awaited_task_returns_no_value);
+                AddSection(QuickInfoSectionKinds.Description, builder.ToImmutable());
+                return QuickInfoItem.Create(token.Span, sections: sections.ToImmutable());
+            }
```

Comment:
i'd personally prefer that all code related to ShowAwaitReturn be cent5ralized.  but that's me :)

Reply:
(optionally in a helper method too).

Reply:
Refactored. It duplicates a bit of other code but yeah it's better: https://github.com/dotnet/roslyn/compare/5205913227e97cb0fcf276b0371a5d23f35af45d..736d3f6a7c00819ee8e2a697ae7b3044922433b4

Helper method won't be done since some of the code there is calling local functions which captures state, and I don't think that's remotely the worst sin happening in that code at this point.